rust/bridge/node/src/lib.rs

```diff
@@ -3,13 +3,289 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+use async_trait::async_trait;
+use libsignal_bridge::node;
+use libsignal_protocol::*;
+use neon::context::Context;
 use neon::prelude::*;
+use signal_neon_futures::*;
+use std::fmt;
+use std::marker::PhantomData;
+use std::panic::AssertUnwindSafe;
+use std::sync::Arc;
 
 pub mod logging;
 
+#[derive(Debug)]
+struct CallbackError {
+    message: String,
+}
+
+impl CallbackError {
+    fn new(message: String) -> CallbackError {
+        Self { message }
+    }
+}
+
+impl fmt::Display for CallbackError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "callback error {}", self.message)
+    }
+}
+
+impl std::error::Error for CallbackError {}
+
+fn js_error_to_rust(func: &'static str, err: String) -> SignalProtocolError {
+    SignalProtocolError::ApplicationCallbackError(func, Box::new(CallbackError::new(err)))
+}
+
+pub fn boxed_object<'a, T: 'static + Send>(
+    cx: &mut TaskContext<'a>,
+    value: T,
+) -> JsResult<'a, JsValue> {
+    Ok(cx.boxed(node::DefaultFinalize(value)).upcast())
```

Comment:
This isn't going to be 100% consistent because mutable objects get stuffed in RefCell first. I recommend using ResultTypeInfo's `convert_into` everywhere.